ccdproc/core.py

```diff
@@ -45,6 +46,161 @@
+        A materbias frame to be subtracted from ccd.
+
+    master_flat: None, `~numpy.ndarray`,  or `~ccdproc.CCDData`
+        A materflat frame to be divided into ccd.
+
+    bad_pixel_mask: None or `~numpy.ndarray`
+        A bad pixel mask for the data. The bad pixel mask should be in given
+        such that bad pixels havea value of 1 and good pixels a value of 0.
+
+    gain: None or `~astropy.Quantity`
+        Gain value to multiple the image by to convert to electrons
+
+    readnoise: None or `~astropy.Quantity`
+        Read noise for the observations.  The read noise should be in
+        electrons.
+
+    oscan_median :  bool, optional
+        If true, takes the median of each line.  Otherwise, uses the mean
+
+    oscan_model :  `~astropy.modeling.Model`, optional
+        Model to fit to the data.  If None, returns the values calculated
+        by the median or the mean.
+
+    min_value : None or float
+        Minimum value for flat field.  The value can either be None and no
+        minimum value is applied to the flat or specified by a float which
+        will replace all values in the flat by the min_value.
+
+
+    Returns
+    -------
+    occd: `~ccdproc.CCDData`
+        Reduded ccd
+
+    Examples
+    --------
+
+    1. To overscan, trim, and gain correct a data set:
+
+    >>> import numpy as np
+    >>> from astropy import units as u
+    >>> from ccdproc import CCDData
+    >>> from ccdproc import ccd_process
+    >>> ccd = CCDData(np.ones([100, 100]), unit=u.adu)
+    >>> nccd = ccd_process(ccd, oscan='[1:10,1:100]', trim='[10:100, 1:100]',\
+                           error=False, gain=2.0*u.electron/u.adu)
+
+
+    """
+    # make a copy of the object
+    nccd = ccd.copy()
+
+    # apply the overscan correction
+    if isinstance(oscan, CCDData):
+        nccd = subtract_overscan(nccd, overscan=oscan,
+                                 median=oscan_median,
+                                 model=oscan_model)
+    elif isinstance(oscan, six.string_types):
+        nccd = subtract_overscan(nccd, fits_section=oscan,
+                                 median=oscan_median,
+                                 model=oscan_model)
+    elif oscan is None:
+        pass
+    else:
+        raise TypeError('oscan is not None, a string, or CCDData object')
+
+    # apply the trim correction
+    if isinstance(trim, six.string_types):
+        nccd = trim_image(nccd, fits_section=trim)
+    elif trim is None:
+        pass
+    else:
+        raise TypeError('trim is not None or a string')
+
+    # create the error frame
+    if error and gain is not None and readnoise is not None:
+        nccd = create_deviation(nccd, gain=gain, readnoise=readnoise)
+    elif error and (gain is None or readnoise is None):
+        raise ValueError(
+            'gain and readnoise must be specified to create error frame')
+
+    # apply the bad pixel mask
+    if isinstance(bad_pixel_mask, np.ndarray):
```

Comment:
I think we are generally assuming we are starting with a fresh frame, but we can revisit this as needed